src/libraries/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/PEHeaders.cs

```diff
@@ -93,13 +93,21 @@ public PEHeaders(Stream peStream, int size, bool isLoadedImage)
             _coffHeaderStartOffset = reader.Offset;
             _coffHeader = new CoffHeader(ref reader);
 
-            if (!isCoffOnly)
+            if (isCoffOnly)
+            {
```

Comment:
```suggestion
            {
                if (isLoadedImage)
                {
                    throw new BadImageFormatException(SR.InvalidPESignature);
                }
```

What do you think about adding this check (a bit earlier)? The OS can't load an object file, can it?

Reply:
Adding such check sounds good to me.